sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeTestUtils.scala

```diff
@@ -54,6 +55,13 @@ object DateTimeTestUtils {
     "Europe/Amsterdam")
   val outstandingZoneIds: Seq[ZoneId] = outstandingTimezonesIds.map(getZoneId)
 
+  private val random = new Random(System.nanoTime())
+
+  // Take 2 samples from the `outstandingZoneIds`. This is useful when the test case is slow.
```

Comment:
This might be too much. There are only 9 special timezones in total, and it seems OK to always test all of them.

Reply:
Originally, I did this for improving the test case `SPARK-34761,SPARK-35889: add a day-time interval to a timestamp`. But I am fine with your comment